text/0000-pack-direct-podman-call.md

```diff
@@ -0,0 +1,86 @@
+# Meta
+[meta]: #meta
+- Name: Pack direct podman call
+- Start Date: 2023-06-16
+- Author(s): dvaumoron
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+A flag `--container-engine` will be added to the `pack` CLI, the allowed flag value will be `docker` (the current functioning) and `podman` to call podman directly (without socket call)
```

Comment:
Can we add a little motivation on why we want to avoid the socket call?

Reply:
This is a way to make pack standalone, developer would be able to use pack without docker or podman, however they will still need a tool like skopeo to move OCI images.

Reply:
> Can we add a little motivation on why we want to avoid the socket call?

`podman` doesn't react the socket by default and it needs to be started by calling `systemd` manually. https://github.com/containers/podman/blob/main/docs/tutorials/socket_activation.md

Probably the socket concern is the security. I don't know if another process can interfere with existing socket communication, but if it can call podman, I guess there is no difference in security.

Reply:
docker act as root, podman doesn't need that, so you can use it on a computer without root access (like a company one), in that case why bother to launch a socket in user space ? Moreover as i have already said, using podman as a library, we can make the pack utility standalone, that could be usefull for some case where you don't run container locally, because you will not need docker or podman at all.